src/js/node/net.ts

```diff
@@ -100,6 +100,11 @@ const Socket = (function (InternalSocket) {
       data({ data: self }, buffer) {
         if (!self) return;
 
+        if (self.#onreadCallback) {
```

Comment:
```suggestion
        const onReadCallback = self.#onreadCallback;
        if (onReadCallback) {
            try {
              onReadCallback(buffer.length, buffer);
            } catch(e) {
              self.emit("error", e);
           }
            return;
        }
```

Reply:
Ideally, we'd have a separate handlers object and not do this check here on every single call to onData

Reply:
Changed to use a separate handlers object